### ISSUE TITLE
VideoCapture() doesn't work (blocks) when running in a separate thread

### ISSUE THREAD

**Issue by gfarrell** [opencv/opencv#7519]
##### System information (version)

- OpenCV => 3.1.0_4 (via `brew install opencv3 --with-python3`)
- OS => OSX Yosemite
- Platform => python3.5.2

##### Detailed description

When using `cv2.VideoCapture(0)` in a thread, it doesn't work, but it does work when running in the main thread. It doesn't raise an Exception, merely blocks until the programme exits.

##### Steps to reproduce

Using this script as a minimum viable example:

```python
from threading import Thread
import cv2
from time import sleep
from PIL import Image

def main():
    print("Running in main thread")
    cam = cv2.VideoCapture(0)
    print("Starting capture")
    sleep(2)
    _, img = cam.read()
    print("Acquired image")
    sleep(2)
    Image.fromarray(img).save("tmp.jpg", "JPEG")
    cam.release()
    print("End process")

th = Thread(target=main, name="Testing Thread")
th.daemon = True
th.start()
th.join()
```

When this is run on OSX Yosemite, the only thing that gets printed out is "Running in main thread". The rest of the `main()` function does not get executed. Stepping into `VideoCapture()` with `pdb` just hangs as well. If the `main()` function is run normally (without threading) then it works fine.

N.B. I originally had this working on macOS Sierra, which I had to install via `brew install opencv3 --with-python3 --HEAD` (note the extra `--HEAD` option) as installing it without `--HEAD` wouldn't work on Sierra. Similarly, installing *with* `--HEAD` does *not* work on OSX Yosemite.

**Comment by alalek** [comment]
Does `VideoCapture` work with file instead of camera device?

**Comment by gfarrell** [comment]
@alalek yes, I tested it with opening a file (`test.mov`) and it worked.

**Comment by lettergram** [comment]
Could not replicate on:

* OpenCV => 3.3.2 (via brew install opencv3 --with-python3)
* OS => OSX 10.11.6 (El Capitan)
* Platform => python3.6

**Comment by brssrm** [comment]
I also experience the same, windows 7, opencv-3.2.0, python 2.7

**Comment by creotiv** [comment]
Same thing, app hanging on exit when i use VideoCapture in thread on a video file.

**Comment by pwnorbitals** [comment]
Is this still a problem on latest releases ?

**Comment by gfarrell** [comment]
I haven't tried, but I provided an MVE above so you could try running it to check.
